### PR TITLE
Use the Flickr API for Flickr

### PR DESCRIPTION
**Pull request by xabolcs**
The old code is still functional, and the [API equivalent](https://www.flickr.com/services/api/flickr.interestingness.getList.html) `flickr.interestingness.getList` _returns the list of interesting photos for the most recent day or a user-specified date._

### CODE REVIEW

**Comment by xabolcs** [comment]
The old code returns photos from the last seven day with one query.
Photos from last seven day needs seven API queries. :-1: 

Should I provide photos from last seven days?

It's OK to me to close this PR without merging at all! :)

File: Flickr/bootstrap.js [hunk 1]
```diff
@@ -19,8 +19,15 @@ function startup(aData, aReason) {
 	
 	xhr.addEventListener('load',function(){
 		if(xhr.status == 200) {
-			let m = xhr.responseText.match(/a data-track="thumb"[^>]+><img src="([^"]+)_m.jpg"/),
-			URL = m && xhr.channel.URI.resolve(m[1]+'_b.jpg') || null;
+			try {
+				let m = JSON.parse(xhr.responseText).photos.photo;
+				m.sort(function() 0.5-Math.random());
+				m = m.shift();
```
Comment (xabolcs) [inline Flickr/bootstrap.js:25]: `flickr.interestingness.getList` returns it's result in the same order.

File: Flickr/bootstrap.js [hunk 2]
```diff
@@ -53,8 +60,20 @@ function startup(aData, aReason) {
...
+	formData.append("extras", "url_l,url_h");
+	formData.append("per_page", 500);
+	xhr.open('POST', 'https://api.flickr.com/services/rest/', true);
+	xhr.send(formData);
+}
+
+function flickerInterestingnessForm () {
+	let formData = Cc['@mozilla.org/files/formdata;1'].createInstance(Ci.nsIDOMFormData);
+	formData.append("method", "flickr.interestingness.getList");
+	formData.append("api_key", "7cef40b1fbba6eccddc67f6263481cff");
```
Comment (xabolcs) [inline Flickr/bootstrap.js:73]: New app, new key.

Comment (diegocr) [inline Flickr/bootstrap.js:73]: Awesome, thanks a lot :)

**Comment by diegocr** [comment]
> Should I provide photos from last seven days?

Looks good as-is, so merging, thank you!

**Pull request merged**